packages/@expo/metro-config/src/ExpoMetroConfig.ts

```diff
@@ -146,7 +143,7 @@ export function getDefaultConfig(
     },
     watcher: {
       // strip starting dot from env files
-      additionalExts: envFiles.map((file: string) => file.replace(/^\./, '')),
+      additionalExts: ['cjs', 'mjs', ...envFiles.map((file: string) => file.replace(/^\./, ''))],
```

Comment:
This would land in SDK 50, where mjs support is already enabled in a more robust way. Backporting would potentially break existing projects due to https://github.com/facebook/react-native/pull/38628

Reply:
The change also doesn't account for [Node.js file handling](https://github.com/expo/expo/blob/d59c12181347ebcaea8c1134a27ed40672daf0f1/packages/%40expo/cli/src/start/server/metro/withMetroMultiPlatform.ts#L224-L226) which requires some reordering. Unclear to me which version of React Native the Metro change was meant to support.